src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -2039,3 +2189,71 @@ bool TheoryStringsRewriter::stripConstantEndpoints(std::vector<Node>& n1,
   //      which is 3.
   return changed;
 }
+
+bool TheoryStringsRewriter::checkEntailArith(Node a, Node b, bool strict)
+{
+  Node diff = NodeManager::currentNM()->mkNode(kind::MINUS, a, b);
+  return checkEntailArith(diff, strict);
+}
+
+bool TheoryStringsRewriter::checkEntailArith(Node a, bool strict)
+{
+  if (a.isConst())
+  {
+    return a.getConst<Rational>().sgn() >= (strict ? 1 : 0);
+  }
+  else
+  {
+    Node ar = strict
+                  ? NodeManager::currentNM()->mkNode(
```

Comment:
Not sure how important this is but we may want to keep in mind that we are losing some cases by treating > 0 uniformly with >= 0. E.g. `str.len(str.itos(x)) > 0` is always true.

Reply:
This will eventually be handled at a higher level by over/under-approximation transformations like t ----> t_l where we know t > t_l or t ----> t_u where t < t_u.  For example, str.len(str.itos(x)), its under-approximation would be 1. 

The part of the code that this sort of reasoning would go I've marked here:
https://github.com/ajreynol/CVC4/blob/stringsRewriting-v2/src/theory/strings/theory_strings_rewriter.cpp#L2218

We may need to revisit whether its the best idea to separate the entailment test from the over/under approximation, but for now I'd like to try to keep them separate.

Reply:
Sounds good to me.